pkg/server/debug/pprofui/server.go

```diff
@@ -263,13 +332,41 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if !isGoPProf {
 		http.Redirect(w, r, origURL.String(), http.StatusTemporaryRedirect)
 	} else {
-		_ = s.storage.Get(id, func(r io.Reader) error {
+		_ = s.storage.Get(id, func(isProtoProfile bool, r io.Reader) error {
 			_, err := io.Copy(w, r)
 			return err
 		})
 	}
 }
 
+// FilterStacksWithLabels writes all the stacks that have pprof labels matching
+// any one of the expectedLabels to resBuf.
+func FilterStacksWithLabels(stacks []byte, labelFilter string) []byte {
+	if labelFilter == "" {
+		return stacks
+	}
+	res := bytes.NewBuffer(nil)
+	goroutines := strings.Split(string(stacks), "\n\n")
+	// The first element is the nodeID which we want to always keep.
+	res.WriteString(goroutines[0])
+	goroutines = goroutines[1:]
+	regex := regexp.MustCompile(fmt.Sprintf(`labels: {.*%s.*}`, labelFilter))
```

Comment:
We've had some problems[^1] with graphviz breaking when label values contained non-alphanumerics, do you think this will be an issue here? Concretely, `"` and `{` `}`. I wonder if it could be more robust to parse what follows `labels: ` via `json.Decode`, but Hopefully `"` and the like are escaped properly.

[^1]: https://github.com/cockroachdb/cockroach/issues/101523

Reply:
Interesting, thanks for calling it out. I added some tests to ensure we handle `"` and `{` and `}`. The regexp seems to correctly match against these non-alphanumeric labels too. Considering goroutines with `debug=1` will only be available in txt format I don't think such labels break any of our logic.